src/expr/core/src/expr/expr_udf.rs

```diff
@@ -260,38 +256,21 @@ pub(crate) fn get_or_create_flight_client(link: &str) -> Result<Arc<ArrowFlightU
 /// Get or create a wasm runtime.
 ///
 /// Runtimes returned by this function are cached inside for at least 60 seconds.
-/// Later calls with the same link will reuse the same runtime.
+/// Later calls with the same binary will reuse the same runtime.
 #[cfg_or_panic(not(madsim))]
-pub async fn get_or_create_wasm_runtime(link: &str) -> Result<Arc<WasmRuntime>> {
-    static RUNTIMES: LazyLock<Cache<String, Arc<WasmRuntime>>> = LazyLock::new(|| {
+pub fn get_or_create_wasm_runtime(binary: &[u8]) -> Result<Arc<WasmRuntime>> {
+    static RUNTIMES: LazyLock<Cache<md5::Digest, Arc<WasmRuntime>>> = LazyLock::new(|| {
         Cache::builder()
             .time_to_idle(Duration::from_secs(60))
             .build()
     });
 
-    if let Some(runtime) = RUNTIMES.get(link).await {
+    let md5 = md5::compute(binary);
+    if let Some(runtime) = RUNTIMES.get(&md5) {
```

Comment:
What about using `get_with` here? BTW, is it possible that there's an `md5` collision? 🤣

Reply:
I didn't know `get_with` before you mentioned it. Looks good to use.
I think the possibility of md5 collision is negligible as this is not a security critical case. 😄

Reply:
I think we have the udf's identifier? Why bother to use md5? 🤔

Reply:
They are different. The identifier is used for finding functions within a WASM binary, while multiple functions may share the same binary. Here we cache the WASM runtime for each binary, but we don't want to store full binaries in memory, so their md5s are used as the cache key.

Reply:
You are right. But my idea is that the case where multiple functions share the same binary is also rare. Anyway, both OK to me.